postgresql/provider.go

```diff
@@ -246,6 +255,32 @@ func getRDSAuthToken(region string, profile string, username string, host string
 		return "", err
 	}
 
+	if role != "" {
+		stsClient := sts.NewFromConfig(awscfg)
+		roleInput := &sts.AssumeRoleInput{
+			RoleArn:         aws.String(role),
+			RoleSessionName: aws.String("TerraformSession"),
```

Comment:
Maybe something like
```suggestion
			RoleSessionName: aws.String("TerraformPostgresqlProvider"),
```
as Terraform may also assume a role for the `aws` provider (and maybe some others), this could allow to identify which provider is using this session?